README.md
# UW-NSL-Web-page-Management-System


JSON/news
[{"news": [{"newsId": 68,"title": "Prof. Conti is visiting NSL","description": "Prof. Mauro Conti is visiting NSL for a week. He is a full professor in the University of Padua, Italy, and an affiliate Professor in the University of Washington, USA. His research interests are mainly in the area of security and privacy. In this area, he published more than 170 papers in topmost international peer-reviewed journals and conferences, including IEEE TIFS, IEEE TDSC, IEEE TPDS, ACM TWEB, ACM/IEEE TON, IEEE TSC, IEEE COMST, ACM CCS, Usenix Security, ACM AsiaCCS, ACM WiSec, ACM SACMAT, ACM MobiHoc, ACNS, IEEE ICDCS, and ESORICS.","date": "2018-07-18","image1": "https://labs.ece.uw.edu//nsl/faculty/radha/news_images/MauroConti_NotMyFinger.jpg","imgDes1": "Prof. Mauro Conti ","image2": "","imgDes2": "","links": "[{\"description\":\"Prof. Mauro Conti's website\",\"link\":\"http://www.math.unipd.it/~conti/index.html\"}]"},{"newsId": 71,"title": "Paper presented at ACC 2018","description": "Co-authored by G. Yang, H. Hosseini, D. Sahabandu, A. Clark, J. Hespanha, and R. Poovendran, the paper \"Modeling and Mitigating the Coremelt Attack\" is presented in the American Control Conference (ACC), Milwaukee, USA, June 2018. ","date": "2018-06-27","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": "[{\"description\":\"ACC2018 Paper: Modeling and Mitigating the Coremelt Attack\", \"link\":\"https://www2.ee.washington.edu/research/nsl/papers/Coremelt.pdf\"}]"},{"newsId": 70,"title": "Prof. Andrew Clark and Prof. Linda Bushnell participated in STEM workshop at ACC 2018","description": "Prof. Andrew Clark, alumni of UW NSL, Assistant Professor in Electrical and Computer Engineering at Worcester Polytechnic Institute, and Prof. Linda Bushnell, Research Professor at UW NSL, participates in STEM workshop at ACC 2018, the 2018 American Control Conference in Milwaukee, WI, USA. The ACC  offers workshops addressing current and future topics in automatic control from experts in academia, national laboratories, and industry.","date": "2018-06-26","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": "[{\"description\":\"ACC2018 Workshops\", \"link\":\"acc2018.a2c2.org/technical-program/workshops/\"}]"},{"newsId": 69,"title": "Xuhang (Shaun) Ying received his PhD!","description": "Xuhang Ying successfully defended his PhD dissertation titled \"Crowdsourcing and Resource Allocation in Shared Spectrum\" and will continue his research at UW NSL as a postdocotral research associate. Congratulations, Dr. Ying!","date": "2018-06-01","image1": "https://labs.ece.uw.edu//nsl/faculty/radha/profile_pics/shaun.jpg","imgDes1": "Dr. Xuhang (Shaun) Ying","image2": "","imgDes2": "","links": "[{\"description\":\"Dr. Ying's website\",\"link\":\"http://students.washington.edu/xhying/\"}]"},{"newsId": 1,"title": "Joint Boeing-UW Aviation Privacy Patent Award","description": "A patent application on location privacy of air traffic communications, jointly filed by Dr. Krishna Sampigethaya (Boeing) and Prof. Radha Poovendran on April 13, 2010, was issued by the USPTO as patent# 8248294 on August 21, 2012.","date": "2012-08-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": null},{"newsId": 2,"title": "Andrew Clark wins WIPOT 2012 Best Paper Award","description": "Andrew Clark has received the best paper award at the Tenth International Symposium on Modeling and Optimization In Mobile, Ad Hoc and Wireless Networks (WiOpt 2012). The title of the paper was \"Leader Selection for Minimizing Convergence Error in Leader-Follower Systems: A Supermodular Optimization Approach,\" and was co-authored by EE faculty advisors Linda Bushnell and Radha Poovendran. Congratulations to Andrew and Professors Bushnell and Poovendran.","date": "2012-05-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": "[{\"description\":\"More information\", \"link\":\"http://wi-opt.cs.upb.de/WiOpt_2012/Home.html\"}]"},{"newsId": 3,"title": "Tony graduates and joins cisco","description": "He (Tony) Wu successfully completed his MS and joined Advanced Technology Group at Cisco. Congratulations and all the best to Tony!","date": "2011-08-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": null},{"newsId": 4,"title": "Sid graduates and joins microsoft","description": "Sidharth Nabar successfully defended his PhD dissertation titled \"Designing Body Sensor Networks for Energy Efficiency and Performance\" and joined the Mobile Phone Group at Microsoft. Congratulations and all the best to Dr. Sid!","date": "2011-08-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": null},{"newsId": 5,"title": "Basel graduates and joins KACST","description": "Basel Alomair successfully defended his PhD dissertation titled \"Towards Authenticated and Private Computer and Wireless Communications\" and joined the Computer Research Institute (CRI) at King Abdulaziz City for Science and Technology (KACST) as a Assistant Research Professor. Congratulations and all the best to Prof. Basel!","date": "2011-08-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": null},{"newsId": 6,"title": "Basel wins the UW EE Outstanding Research Award","description": "Basel Alomair has won the UW Electrical Engineering Outstanding Research Award, recognizing his outstanding contributions to research in the department.","date": "2011-05-01","image1": null,"imgDes1": null,"image2": null,"imgDes2": null,"links": null},{"newsId": 7,"title": "Boeing-NSL paper receives Best Paper Award at IEEE/AIAA DASC","description": "Boeing-NSL paper by Krishna Sampigethaya and Radha Poovendran, titled \"Visualization and Assessment of ADS-B Security for Green ATM,\" has been recognized with the Best Paper Award for the NextGen Surveillance Session at the 29th IEEE/AIAA Digital Avionics Systems Conference (DASC). The conference is being held at Salt Lake City, UT, from October 3-7, 2010. The pa

src/webpageeditor/SCP.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import com.jcraft.jsch.ChannelSftp;
import com.jcraft.jsch.JSch;
import com.jcraft.jsch.Session;
import java.awt.Desktop;
import java.io.File;
import java.net.URI;
import java.net.URISyntaxException;
import java.net.URL;
import javax.swing.JOptionPane;

/**
 *
 * @author kalan
 */
public class SCP {

    private String UserName="";
    private String Password="";
    private String host = "labs.ece.uw.edu";
    private String destination = "/var/www/html/research/nsl/faculty/radha/assets/json/";
    private boolean loginPassed = false;
    private JSch jsch=new JSch();
         
    public boolean connectTest(String username, String password ) 
    {
        System.out.println("Running SCP");
         try
        {


          Session session=jsch.getSession(username, host, 22);

          java.util.Properties config = new java.util.Properties(); 
          config.put("StrictHostKeyChecking", "no");
          session.setPassword(password);
          session.setConfig(config);
          session.connect();
          session.disconnect();
          
          //keeping a copy for future use
          UserName = username;
          Password=password;
          loginPassed = true;
          
        }catch (Exception x)
        {
            System.err.println("failed");
            return false;

        }
	 
         return true;
    }
    
    
    public void uploadFile(String fileName)
    {
        try
        {
            Session session = jsch.getSession(UserName, host);
            session.setPassword(Password);
            session.connect();
            ChannelSftp sftpChannel = (ChannelSftp) session.openChannel("sftp");
            sftpChannel.connect();
            sftpChannel.put(System.getProperty("user.dir")+"\\JSON\\"+fileName, destination+fileName,ChannelSftp.OVERWRITE);
            sftpChannel.disconnect();
            session.disconnect();
            

        }catch (Exception x)
        {
            JOptionPane.showMessageDialog(null, "Failed to commit \""+fileName+"\" file. Please try again !","Commit failed !", JOptionPane.ERROR_MESSAGE);
        }
        
        
        JOptionPane.showMessageDialog(null, "Web Page has been successfully updated !","Web Page Successfully Updated!", JOptionPane.INFORMATION_MESSAGE);
        
    }
    
    
    public void uploadFliles(String filePath,String destination_) //destination always end with a /
    {
        try
        {
            Session session = jsch.getSession(UserName, host);
            session.setPassword(Password);
            session.connect();
            ChannelSftp sftpChannel = (ChannelSftp) session.openChannel("sftp");
            sftpChannel.connect();
            sftpChannel.put(filePath, destination_+new File(filePath).getName(),ChannelSftp.OVERWRITE);
            sftpChannel.disconnect();
            session.disconnect();
        }catch (Exception x)
        {
            JOptionPane.showMessageDialog(null, "Failed to upload selected file. Please try again !","Commit failed !", JOptionPane.ERROR_MESSAGE);
        }
        
        
        
    }
    
    public boolean getLoginStatus()
    {
        return loginPassed;
    }
    
    
    public String getUserName()
    {
        return UserName;
    }
    
    public static boolean openWebpage(URI uri) {
    Desktop desktop = Desktop.isDesktopSupported() ? Desktop.getDesktop() : null;
    if (desktop != null && desktop.isSupported(Desktop.Action.BROWSE)) {
        try {
            desktop.browse(uri);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    return false;
}

public static boolean openWebpage(URL url) {
    try {
        return openWebpage(url.toURI());
    } catch (URISyntaxException e) {
        e.printStackTrace();
    }
    return false;
}
    
            
    

}


src/webpageeditor/fillTable.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import java.awt.Font;
import java.util.HashMap;
import java.util.List;
import javax.swing.JTable;
import javax.swing.JTextArea;
import javax.swing.table.DefaultTableModel;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

/**
 *
 * @author Kalana
 */
public class fillTable {
    
    //this will change tables font size
    public void initializetables(JTable[] table)
    {
        for(int loop=0;loop<table.length;loop++)
        {
            
           
            //make their cell uneditable
            //set table header size and font size
            table[loop].getTableHeader().setFont(new Font("Dialog",Font.PLAIN,15));
            //setting row sizes and font sizes
            table[loop].setFont(new Font("Dialog",Font.PLAIN,15));
            table[loop].setRowHeight(30);
        }
    }
    
    public void completeTable(JTable table,List<String> Columns, List<HashMap<String,String>> rowData)
    {
        
        System.out.println(rowData);
        Object row[] = new Object[Columns.size()];
        DefaultTableModel tmodel = (DefaultTableModel)table.getModel();
        //lets make sure we clear all the data in the table before we add new
        tmodel.setRowCount(0);
        
        for(int loop=0;loop<rowData.size();loop++)
        {
            for(int inner=0;inner<Columns.size();inner++)
                row[inner]= rowData.get(loop).get(Columns.get(inner));
            
           tmodel.addRow(row);
        }
        
    }
    
    //this will take given text areas and fill them with user clicked rows data
    public void fillTextAreas(JTable table,JTextArea areas[])
    {
        DefaultTableModel myModel = (DefaultTableModel)table.getModel();
        for(int loop=0;loop<areas.length;loop++)
        {
            areas[loop].setText((String)myModel.getValueAt(table.getSelectedRow(),loop+1));
        }
        System.err.println(table.getSelectedRow());
    }
    
    //this clear values from given text areas
    public void clearFields(JTextArea areas[])
    {
        for(int loop=0;loop<areas.length;loop++)
        {
            areas[loop].setText("");
        }
    }
    
    
    public void fillNewsLinkTable(JTable table, String JSONstring,String[] columns)
    {
        DefaultTableModel mymodel = (DefaultTableModel)table.getModel();
        JSONParser parser = new JSONParser(); 
        JSONArray a = null;
        //removing data every time
        mymodel.setRowCount(0);
        
        if(JSONstring == null || JSONstring.equals(""))
            return;
        
            try
            {
                a = (JSONArray)parser.parse(JSONstring);
                Object[] rows = new Object[columns.length];

                for(int loop=0;loop<a.size();loop++)
                {
                    for(int inner=0;inner<columns.length;inner++)
                        rows[inner] = ((JSONObject)a.get(loop)).get(columns[inner]);

                    mymodel.addRow(rows);
                }

            }catch (Exception x)
            {
                x.printStackTrace();
            }
   }
    

}


src/webpageeditor/JSONparser.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

/**
 *
 * @author Kalana
 */
public class JSONparser {
        
        
        
    //this will parse the user given JSON file according to link and the given keys and return a key and value pair
    public List<HashMap<String, String>> parser(List<String> keys, String Link, String ArrayName)
    {
        List<HashMap<String,String>> ListOmaps= new ArrayList<HashMap<String,String>>();
        HashMap<String, String> values = new HashMap<>();
        
        JSONParser parser = new JSONParser(); 
        JSONArray a = null;
        try {

            a = (JSONArray)parser.parse(new FileReader(System.getProperty("user.dir")+"\\JSON\\"+Link));
            JSONObject obj = (JSONObject)a.get(0);
            // Loop through each item
                
            for(int currSize=0;currSize<((JSONArray)obj.get(ArrayName)).size();currSize++)
            {
                JSONObject obj_ = (JSONObject)((JSONArray)obj.get(ArrayName)).get(currSize);
                for(int currElement=0;currElement<keys.size();currElement++)
                {
                    String val = null;
                    
                    if(obj_.get(keys.get(currElement)) instanceof Long) //takes care of Long to String casting error
                        val = Long.toString((Long)obj_.get(keys.get(currElement)));
                    else
                        val = (String)obj_.get(keys.get(currElement));
                    
                    values.put(keys.get(currElement), val);
                    
                    
                }
                ListOmaps.add(new HashMap<String, String>(values));
       
            }

        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } catch (ParseException e) {
            e.printStackTrace();
        }
        
        System.err.println("return List size = "+ListOmaps.size());
        return ListOmaps;
    }
    
    
}


src/webpageeditor/fetchFiles.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardCopyOption;
import java.util.List;
import javax.swing.JOptionPane;

/**
 *
 * @author kalan
 */
public class fetchFiles {
    
    public void downloadFiles(String WebstiteUrl, List<String>files) throws MalformedURLException, IOException
    {
        for(int loop=0;loop<files.size();loop++)
        {
            Path target = Paths.get(System.getProperty("user.dir")+"\\JSON\\"+files.get(loop));
            URL website = new URL(WebstiteUrl+files.get(loop));

            try (InputStream in = website.openStream())
            {
                Files.copy(in, target, StandardCopyOption.REPLACE_EXISTING);
            }catch (Exception x)
            {
                JOptionPane.showMessageDialog(null, "Files cannot be fetched due to invalid URL.","URL not found", JOptionPane.ERROR_MESSAGE);
            }
        }
        
    
    }
}


src/webpageeditor/fileOperations.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import javax.swing.JOptionPane;
import javax.swing.JTable;
import javax.swing.JTextArea;
import javax.swing.table.DefaultTableModel;

/**
 *
 * @author kalan
 */
public class fileOperations {
    public void addData(JTable table,String fileLocation, String header, String[] ColumnNames, JTextArea[] textboxes) 
    {
        int loop=0;
        DefaultTableModel myModel = (DefaultTableModel)table.getModel();
        ArrayList<String> possibleIDs = new ArrayList<String>();
        //just a save procedure to make sure we dont repeat IDs 
        for(loop=0;loop<table.getRowCount();loop++)
            possibleIDs.add((String)myModel.getValueAt(loop, 0));
        
        int nextId =0;
        
        if(table.getRowCount() == 0)
            nextId = 1;
        else
            nextId = Integer.parseInt((String)myModel.getValueAt(loop-1, 0));
        
        while(true)
        {
            if(!possibleIDs.contains(Integer.toString(nextId)))
                break;
                
            nextId++;   
        }
        //at this poin we have a unique id
        Object[] row = new Object[textboxes.length+1];
        row[0]= Integer.toString(nextId); //adding ID
        for(int curr=1;curr<row.length;curr++)
            row[curr]=textboxes[curr-1].getText();

        myModel.addRow(row);
        
        
        writetoFile(JSONmaker(table,header,ColumnNames), fileLocation);
    }
    
    //edit button logic
    public void editData(JTable table,String fileLocation, String header, String[] ColumnNames, JTextArea[] textboxes) 
    {
        //if table is empty theres nothing to delete
        if(table.getRowCount() == 0)
            return;
        
        if(table.getSelectedRow() == -1)
        {
             JOptionPane.showMessageDialog(null, "Please select a row first.","Row not selected", JOptionPane.ERROR_MESSAGE);
             return;
        }
        
        DefaultTableModel myModel = (DefaultTableModel)table.getModel();
        for(int loop=1;loop<table.getColumnCount();loop++)
            myModel.setValueAt(textboxes[loop-1].getText(), table.getSelectedRow(), loop);
        
        writetoFile(JSONmaker(table,header,ColumnNames), fileLocation);
    }
    public void removeData(JTable table,String fileLocation, String header, String[] ColumnNames) 
    {
        //if table is empty theres nothing to delete
        if(table.getRowCount() == 0)
            return;
        
        if(table.getSelectedRow() == -1)
        {
             JOptionPane.showMessageDialog(null, "Please select a row first.","Row not selected", JOptionPane.ERROR_MESSAGE);
             return;
        }
            
        DefaultTableModel myModel = (DefaultTableModel)table.getModel();
        myModel.removeRow(table.getSelectedRow());
        
        
        
        if(table.getRowCount() == 0)
        {
            switch(header)
            {
                case "links" :  writetoFile("[{\"links\":[]},200]" , fileLocation);
                                return;
                                
                case "research" : writetoFile("[{\"research\":[]},200]" , fileLocation);
                                  return;  
                
                case "publications" : writetoFile("[{\"publications\":[]},200]" , fileLocation);
                                  return;
                                  
                case "tools" : writetoFile("[{\"tools\":[]},200]" , fileLocation);
                                  return;
                                  
                case "courses" : writetoFile("[{\"courses\":[]},200]" , fileLocation);
                                  return;
                                  
                case "people" : writetoFile("[{\"people\":[]},200]" , fileLocation);
                                  return;
                                  
                case "news" : writetoFile("[{\"news\":[]},200]" , fileLocation);
                                  return;                    
            }
            
        }
        

        writetoFile(JSONmaker(table,header,ColumnNames), fileLocation);

    }
    
    //this will create a JSON string according to the given jtable
    private String JSONmaker(JTable table,String header,String[] ColumnNames)
    {
        DefaultTableModel myModel = (DefaultTableModel)table.getModel();
        String JSONstring="[{\""+header+"\": [";
        String JSONpacket ="";
        
        for(int loop=0;loop<table.getRowCount();loop++)
        {
            JSONpacket = JSONpacket+"{" ;
            for(int innerLoop=0;innerLoop<ColumnNames.length;innerLoop++)
            {
                
                
                if(innerLoop == 0)
                    JSONpacket = JSONpacket + "\""+ ColumnNames[innerLoop]+"\": " +specialCharacterHandler((String)myModel.getValueAt(loop,innerLoop))+",";
                else
                    if(specialCharacterHandler((String)myModel.getValueAt(loop,innerLoop)) != null)
                        JSONpacket = JSONpacket + "\""+ ColumnNames[innerLoop]+"\": "+"\"" +specialCharacterHandler((String)myModel.getValueAt(loop,innerLoop))+"\",";
                    else
                        JSONpacket = JSONpacket + "\""+ ColumnNames[innerLoop]+"\": "+"" +specialCharacterHandler((String)myModel.getValueAt(loop,innerLoop))+",";
            }
            JSONpacket = removeLastChar(JSONpacket);
            JSONpacket = JSONpacket+"},";
            
        }
        JSONpacket = removeLastChar(JSONpacket);
        JSONpacket = JSONpacket + "]}, 200]";
        JSONstring = JSONstring+JSONpacket;
        
        return JSONstring;
    }
    
    private void writetoFile(String JSONstring, String fileLocation)
    {
        try
        {
            File myFile = new File(System.getProperty("user.dir")+"\\JSON\\"+fileLocation);
            FileWriter  mywriter = new FileWriter(myFile,false); //overwrite the file
            BufferedWriter buffer = new BufferedWriter(mywriter);
            buffer.write(JSONstring);
            buffer.close();
            
        }catch (IOException x)
        {
             JOptionPane.showMessageDialog(null, fileLocation+" cannot be found under"+System.getProperty("user.dir")+"\\JSON"+" directory","File not found", JOptionPane.ERROR_MESSAGE);
        }
        
    }
    
    
    public String removeLastChar(String phrase)
    {
        String rephrase = null;
        if (phrase != null && phrase.length() > 1)
            rephrase = phrase.substring(0, phrase.length() - 1);


            return rephrase;
    }
    
    
    public String specialCharacterHandler(String value)
    {
        String temp= null;
        //handling " , ' and \
        if(value != null)
        {
            //.replaceAll("\\\\", "\\\\\\\\")
            temp = value.replaceAll("\\\\", "\\\\\\\\").replaceAll("\"", "\\\\\"").replaceAll("\n", "\\\\n").replaceAll("\t", "\\\\t").replaceAll("\b", "\\\\b").replaceAll("\r", "\\\\r").replaceAll("\f", "\\\\f");
            return temp;
        }
        else
            return temp;
        
    }
    
    
    
    

}


src/webpageeditor/login.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webpageeditor;

import com.alee.laf.WebLookAndFeel;
import java.awt.Color;
import java.awt.Dimension;
import java.awt.Toolkit;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import javax.swing.ImageIcon;
import javax.swing.JOptionPane;
import javax.swing.UIManager;

/**
 *
 * @author kalan
 */
public class login extends javax.swing.JFrame {

    /**
     * Creates new form logign
     */
    
    private static String URL = "https://labs.ece.uw.edu/nsl/faculty/radha/assets/json/";
    private static String[] fileNames = new String[]{"alums","courses","currentMembers","facultyMembers","links","news","publications","research","tools","visiting_profs","visiting_studs"};
    private static List<String> downloadFiles = new ArrayList<String>();
    private SCP myscp = new SCP();
    

    public login() {
        initComponents();
        setIconImage(Toolkit.getDefaultToolkit().getImage(getClass().getResource("nsl_webLogo.png")));
        Dimension dim = Toolkit.getDefaultToolkit().getScreenSize();
        this.setLocation(dim.width/2-this.getSize().width/2, dim.height/2-this.getSize().height/2);
        
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jLabel1 = new javax.swing.JLabel();
        username = new javax.swing.JTextField();
        jLabel2 = new javax.swing.JLabel();
        jButton1 = new javax.swing.JButton();
        pass = new javax.swing.JPasswordField();
        jLabel3 = new javax.swing.JLabel();
        notice = new javax.swing.JLabel();

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("NSL Web Page Management System Login");
        setResizable(false);

        jLabel1.setFont(new java.awt.Font("Dialog", 1, 14)); // NOI18N
        jLabel1.setText("User Name");

        username.setFont(new java.awt.Font("Dialog", 0, 15)); // NOI18N
        username.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                usernameActionPerformed(evt);
            }
        });

        jLabel2.setFont(new java.awt.Font("Dialog", 1, 14)); // NOI18N
        jLabel2.setText("Password");

        jButton1.setFont(new java.awt.Font("Dialog", 1, 14)); // NOI18N
        jButton1.setText("Login");
        jButton1.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent evt) {
                jButton1MouseClicked(evt);
            }
        });
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });

        pass.setFont(new java.awt.Font("Dialog", 0, 15)); // NOI18N

        jLabel3.setIcon(new javax.swing.ImageIcon(getClass().getResource("/webpageeditor/images/img.png"))); // NOI18N

        notice.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        notice.setText("* UW credentials required to login.");

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, layout.createSequentialGroup()
                .addContainerGap(444, Short.MAX_VALUE)
                .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 101, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addGap(18, 18, 18))
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(jLabel3, javax.swing.GroupLayout.PREFERRED_SIZE, 138, javax.swing.GroupLayout.PREFERRED_SIZE)
                        .addGap(12, 12, 12)
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.TRAILING, false)
                            .addComponent(jLabel2, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                            .addComponent(jLabel1, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
                        .addGap(44, 44, 44)
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                            .addComponent(username)
                            .addComponent(pass, javax.swing.GroupLayout.DEFAULT_SIZE, 256, Short.MAX_VALUE)))
                    .addComponent(notice, javax.swing.GroupLayout.PREFERRED_SIZE, 525, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addGap(34, 34, 34)
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                            .addComponent(jLabel1, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                            .addComponent(username, javax.swing.GroupLayout.DEFAULT_SIZE, 35, Short.MAX_VALUE))
                        .addGap(18, 18, 18)
                        .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                            .addComponent(jLabel2, javax.swing.GroupLayout.DEFAULT_SIZE, 35, Short.MAX_VALUE)
                            .addComponent(pass)))
                    .addGroup(layout.createSequentialGroup()
                        .addGap(12, 12, 12)
                        .addComponent(jLabel3, javax.swing.GroupLayout.PREFERRED_SIZE, 133, javax.swing.GroupLayout.PREFERRED_SIZE)))
                .addGap(8, 8, 8)
                .addComponent(notice, javax.swing.GroupLayout.PREFERRED_SIZE, 30, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addGap(2, 2, 2)
                .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 45, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void usernameActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_usernameActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_usernameActionPerformed

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_jButton1ActionPerformed

    private void jButton1MouseClicked(java.awt.event.MouseEvent evt) {//GEN-FIRST:event_jButton1MouseClicked
       
    //myscp.connectTest("sdinuka", "$$$131Nisha", "www2.ee.washington.edu");
    if(!myscp.connectTest(username.getText(), pass.getText()))
    {
        notice.setText("* Invalid user name/password or you don't have permission to UW ee servers");
        notice.setForeground(Color.RED);
        pass.setText("");
        
    }
    else
    {
        this.dispose();
        runEditor();
    }
        

    }//GEN-LAST:event_jButton1MouseClicked

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(WebLookAndFeel.class.getCanonicalName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        
 
        //</editor-fold>
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {

                new login().setVisible(true);
            }
        });
    }
    
    
    private void runEditor() 
    {
            //setting up List of file names
            downloadFiles.addAll(Arrays.asList(fileNames));
            fetchFiles files = new fetchFiles();
            
            try
            {
                files.downloadFiles(URL, downloadFiles);
                MainWindow window = new MainWindow();
                //setting up scp 
                window.setSCP(myscp);
                window.setVisible(true);
            }catch (Exception x)
            {
                JOptionPane.showMessageDialog(null, "Internal error has occured please contact the developer !","Fatal Error", JOptionPane.ERROR_MESSAGE);

            }
            

    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton jButton1;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel notice;
    private javax.swing.JPasswordField pass;
    private javax.swing.JTextField username;
    // End of variables declaration//GEN-END:variables
}
